src/components/NewCommentForm.jsx

```diff
@@ -0,0 +1,27 @@
+import { HiUser } from "react-icons/hi2";
+import { postNewCommentByArticleId } from "../../utils/utils"
+import '../css/new_comment_form.css'
+
+export default function NewCommentForm({article_id, newComment, setNewComment, listOfComments, setListOfComments}) {
+    const handleNewComment = (e) => {
+            e.preventDefault();
+            postNewCommentByArticleId(article_id, newComment)
+                .then(({comment}) => {
+                    setNewComment({username: "", body: ""});
+                    setListOfComments((listOfComments) => {
+                        return [comment, ...listOfComments]
+                    })
+                })
+    }
+
+    return (
+        <form className="new_comment" role="new_comment_form" onSubmit={handleNewComment}>
+            <div className="avatar" role="avatar"><HiUser /></div>
+            <label htmlFor="username" className="username_label required">Username: </label>
```

Comment:
Please try to implement some kind of login functionality where a user can be selected from a list and logged in. Or you can have a logged in user by default. 

At the moment we don't know what users are available and we to guess and type which is not ideal in this case.